File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/playback/view/nowplaying/storage/INowPlayingRepository.java
package com.lasthopesoftware.bluewater.client.playback.view.nowplaying.storage;

import com.namehillsoftware.handoff.promises.Promise;

public interface INowPlayingRepository {
	Promise<NowPlaying> getNowPlaying();

	Promise<NowPlaying> updateNowPlaying(NowPlaying nowPlaying);
}

File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/playback/nowplaying/storage/INowPlayingRepository.kt
package com.lasthopesoftware.bluewater.client.playback.nowplaying.storage

import com.namehillsoftware.handoff.promises.Promise

interface INowPlayingRepository {
    fun promiseNowPlaying(): Promise<NowPlaying?>
    fun updateNowPlaying(nowPlaying: NowPlaying): Promise<NowPlaying>
}
